scripts/create_polygons.py

```diff
@@ -46,43 +47,37 @@ def main() -> None:
     output_files = []
 
     for file in source:
-        if not is_tiff(file):
-            get_log().trace("create_polygon_file_not_tiff_skipped", file=file)
+        try:
+            if not is_tiff(file):
+                get_log().trace("create_polygon_file_not_tiff_skipped", file=file)
+                continue
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                source_file_name = os.path.basename(file)
+                tmp_file = os.path.join(tmp_dir, "temp.tif")
+
+                # Get the file
+                write(tmp_file, read(file))
+                file = tmp_file
+
+                # Run create_mask
+                mask_file = os.path.join(tmp_dir, "mask.tif")
+                create_mask(file, mask_file)
+
+                # Run create_polygon
+                data_px_count = get_pixel_count(mask_file)
+                if data_px_count == 0:
+                    # exclude extents if tif is all white or black
+                    get_log().debug(f"- data_px_count was zero in create_mask function for the tif {mask_file}")
+                else:
+                    destination_file_name = os.path.splitext(source_file_name)[0] + ".geojson"
+                    temp_file_path = os.path.join(tmp_dir, destination_file_name)
+                    polygonize_command = f'gdal_polygonize.py -q "{mask_file}" "{temp_file_path}" -f GeoJSON'
+                    os.system(polygonize_command)
+
+                output_files.append(temp_file_path)
+        except ClientError as ce:
+            get_log().error("create_polygon_skip_file", error=str(ce))
```

Comment:
we should fail the command if something fails?

Reply:
This one is interesting. Let's say it fails on reading the file from its source which is a s3 bucket. If for some reasons, some other files are on another bucket that we have access too, should we stop them to be processed?

Reply:
I would process them all and just fail the command at the end

Reply:
Meaning `continue` to the next file when one fails, keep the information that it failed and at the end fail the command if at least one failed?

Reply:
This should not be part of this PR

Reply:
this should be part of the PR, this PR changes the logic from exiting properly with a error code if something goes wrong to swallowing the errors.